### PR TITLE
AuthButton 클릭시 효과 추가

### PR DESCRIPTION
**Pull request by chepchep2**
AuthButton 클릭시 효과를 추가했습니다.
리펙토링이 필요한데 아직 어떻게 접근할지 모르겠어서 하지 못했습니다..

### CODE REVIEW

**Comment by chepchep2** [comment]
* **#33** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/33?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#32** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/32?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#31** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/31?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#30** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/30?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#29** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/29?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#28** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/28?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#27** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/27?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#26** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/26?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/25?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#24** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/24?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/23?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#22** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/22?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/21?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#20** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/20?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#19** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/19?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#18** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/18?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `feature/introPage`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @chepchep2 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: lib/common/auth_button.dart [hunk 1]
```diff
@@ -21,18 +22,49 @@ class AuthButton extends StatelessWidget {
...
   @override
   Widget build(BuildContext context) {
+    final currentBorderColor =
+        widget.borderColor?.withOpacity(0.8) ?? AppColor.transparent;
+    final currentTextColor = widget.textColor ?? AppColor.graysBlack;
+    final currentIconColor = widget.iconColor ?? currentTextColor;
+
     return GestureDetector(
-      onTap: onClick,
+      onTapDown: (_) {
```
Comment (flowkater) [inline lib/common/auth_button.dart:40]: onTap 이 TapDown, TapUp 다 포함입니다. TapDown/TapUp은 지워주세요.

File: lib/common/auth_button.dart [hunk 2]
```diff
@@ -21,18 +22,49 @@ class AuthButton extends StatelessWidget {
...
+          isPressed = true;
+        });
+      },
+      onTapUp: (_) {
+        setState(() {
+          isPressed = false;
+        });
+        widget.onClick();
+      },
+      onTap: () {
```
Comment (flowkater) [inline lib/common/auth_button.dart:51]: onClick 을 받아서 onTap 에서 같이 실행할거라서 widget.onClick 여기로

File: lib/common/auth_button.dart [hunk 3]
```diff
@@ -21,18 +22,49 @@ class AuthButton extends StatelessWidget {
...
+  @override
+  State<AuthButton> createState() => _AuthButtonState();
+}
+
+class _AuthButtonState extends State<AuthButton> {
+  bool isPressed = false;
+
   @override
   Widget build(BuildContext context) {
+    final currentBorderColor =
```
Comment (flowkater) [inline lib/common/auth_button.dart:34]: color 를 여기다가 별도 정의하는 것보다 위젯에 두는게 나을 것 같습니다

File: lib/common/auth_button.dart [hunk 4]
```diff
@@ -21,18 +22,49 @@ class AuthButton extends StatelessWidget {
     this.iconColor,
   });
 
+  @override
+  State<AuthButton> createState() => _AuthButtonState();
+}
+
+class _AuthButtonState extends State<AuthButton> {
+  bool isPressed = false;
```
Comment (flowkater) [inline lib/common/auth_button.dart:30]: elevated Button 처럼 버튼을 한번더 감싸는 위젯을 만드는 게 어떨까요?

**Review by flowkater — CHANGES_REQUESTED** [review]
위젯 리팩토링 시간내서 같이 한번 보시죠

File: lib/common/auth_button.dart (hunk 1 above)
Comment (chepchep2) [reply lib/common/auth_button.dart:40]: 이 부분에 대해서 많이 생각을 해봤습니다만 해결점을 찾지 못해 이렇게 작성했습니다.
onTap 하나만 하면 isPressed의 상태를 어떻게 변경해야할까요??

**Comment by chepchep2** [comment]
### Merge activity

* **Mar 17, 8:12 PM EDT**: @chepchep2 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/23).
* **Mar 17, 8:26 PM EDT**: [Graphite](https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/23) rebased this pull request as part of a merge.
* **Mar 17, 8:27 PM EDT**: @chepchep2 merged this pull request with [Graphite](https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/23).

**Pull request merged**